### PR TITLE
chore(react-charts): fixed linting errors

### PR DESCRIPTION
**Pull request by evwilkin**
What: Towards #3592 - resolves linting errors for react-charts

### CODE REVIEW

File: packages/patternfly-4/react-charts/src/components/ChartBullet/utils/chart-bullet-size.ts [hunk 1]
```diff
@@ -18,6 +18,35 @@ interface ChartBulletScaleInterface {
...
+    ? value + (width - defaultSize) * scale
+    : value - (defaultSize - width) * scale;
+
+// Scale bar width per the given size properties
+export const scaleBarWidth = (props: ChartBulletScaleInterface) => Math.max(scaleDimensions(props), 0);
+
+// Scale size per the given size properties
+export const scaleSize = ({ value, ...rest }: ChartBulletScaleInterface) =>
+  Math.round(
+    scaleDimensions({
```
Comment (dlabrecq) [inline packages/patternfly-4/react-charts/src/components/ChartBullet/utils/chart-bullet-size.ts]: Is it necessary to refactor this?

File: packages/patternfly-4/react-charts/src/components/ChartBullet/utils/chart-bullet-data.ts [hunk 2]
```diff
@@ -197,16 +192,16 @@ export const getQualitativeRangeData = ({
 
   const computedData = datum
     .sort((a: any, b: any) => (invert ? b._y - a._y : a._y - b._y))
-    .map((dataPoint: any, index: number) => {
-      return {
+    .map(
+      (dataPoint: any, index: number) => ({
```
Comment (dlabrecq) [inline packages/patternfly-4/react-charts/src/components/ChartBullet/utils/chart-bullet-data.ts]: Is this required by the linter?

File: packages/patternfly-4/react-charts/src/components/ChartBullet/utils/chart-bullet-theme.ts [hunk 3]
```diff
@@ -27,9 +27,9 @@ interface ChartBulletThemeInterface {
 const getLegendColorScale = (computedData: any, legendData: any) => {
   const colorScale: string[] = [];
   legendData.forEach((data: any, index: number) => {
-    for (let i = 0; i < computedData.length; i++) {
-      if (index === computedData[i]._index) {
-        colorScale.push(computedData[i]._color);
+    for (const i of computedData) {
+      if (index === i._index) {
```
Comment (dlabrecq) [inline packages/patternfly-4/react-charts/src/components/ChartBullet/utils/chart-bullet-theme.ts]: Where are you getting the `index` from? Wouldn't this have to be something like `for (const [i, v] of`?

File: packages/patternfly-4/react-charts/src/components/ChartDonutUtilization/ChartDonutUtilization.tsx [hunk 4]
```diff
@@ -496,11 +509,6 @@ export const ChartDonutUtilization: React.FunctionComponent<ChartDonutUtilizatio
     return computedData;
   };
 
-  const getData = () => {
-    const datum = [{ ...data }];
-    return Data.formatData(datum, { x, y, ...rest }, ['x', 'y']).sort((a: any, b: any) => a._y - b._y);
-  };
-
```
Comment (dlabrecq) [inline packages/patternfly-4/react-charts/src/components/ChartDonutUtilization/ChartDonutUtilization.tsx]: Why are functions being moved? They were in alphabetical order

File: packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts [hunk 5]
```diff
@@ -51,29 +43,15 @@ const getInteractiveLegendItems = ({ chartNames, omitIndex }: ChartInteractiveLe
...
-      type: 'eyeSlash'
-    }
-  };
-};
-
 // Returns targeted events for legend 'data' or 'labels'
 const getInteractiveLegendTargetEvents = ({
   chartNames,
-  isHidden = (index) => false,
+  isHidden = () => false,
```
Comment (dlabrecq) [inline packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts]: This function requires an index

File: packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts [hunk 6]
```diff
@@ -51,29 +43,15 @@ const getInteractiveLegendItems = ({ chartNames, omitIndex }: ChartInteractiveLe
...
   legendName,
-  onLegendClick = (props) => null,
+  onLegendClick = () => null,
```
Comment (dlabrecq) [inline packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts]: This function requires props, provided by Victory

File: packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts [hunk 7]
```diff
@@ -51,29 +43,15 @@ const getInteractiveLegendItems = ({ chartNames, omitIndex }: ChartInteractiveLe
...
   target
 }: ChartInteractiveLegendExtInterface) => {
   if (chartNames === undefined || legendName === undefined) {
-    console.error('getInteractiveLegendTargetEvents:', 'requires chartNames and legendName to be specified');
```
Comment (dlabrecq) [inline packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts]: This is kinda important information to help setup interactive legends properly. Why remove it?

File: packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts [hunk 8]
```diff
@@ -51,29 +43,15 @@ const getInteractiveLegendItems = ({ chartNames, omitIndex }: ChartInteractiveLe
...
-  return !hidden ? {} : {
-    labels: {
-      fill: chart_color_black_500.value
-    },
-    symbol: {
-      fill: chart_color_black_500.value,
-      type: 'eyeSlash'
-    }
-  };
-};
```
Comment (dlabrecq) [inline packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts]: Why are these functions being refactored? What's wrong with leaving this here?

File: packages/patternfly-4/react-charts/src/components/ChartUtils/chart-legend.ts [hunk 9]
```diff
@@ -219,58 +141,32 @@ export const getLegendItemsPerRow = ({
...
-      return getBulletLegendY(rest);
-    default:
-      return getChartLegendY(rest);
+// Returns an approximation of over-sized text width due to growing character count
+//
+// See https://github.com/FormidableLabs/victory/issues/864
+const getTextSizeWorkAround = ({ legendData, legendOrientation, theme }: ChartLegendTextSizeInterface) => {
+  const style = theme.legend.style.labels;
+  if (!(legendData && legendData.length)) {
+    return 0;
```
Comment (dlabrecq) [inline packages/patternfly-4/react-charts/src/components/ChartUtils/chart-legend.ts]: Moving these functions is unnecessary

File: packages/patternfly-4/react-charts/src/components/ChartUtils/chart-theme.ts [hunk 10]
```diff
@@ -128,76 +198,6 @@ export const getDonutUtilizationTheme = (themeColor: string, themeVariant: strin
...
-  switch (themeVariant) {
-    case ChartThemeVariant.dark:
-      return merge(baseTheme, getDarkThemeColors(themeColor));
-    case ChartThemeVariant.light:
-      return merge(baseTheme, getLightThemeColors(themeColor));
-    default:
-      return merge(baseTheme, getLightThemeColors(themeColor));
-  }
-};
-
```
Comment (dlabrecq) [inline packages/patternfly-4/react-charts/src/components/ChartUtils/chart-theme.ts]: Why was moving these functions necessary? They were in alphabetical order

File: packages/patternfly-4/react-charts/src/components/ChartBullet/utils/chart-bullet-data.ts (hunk 2 above)
Comment (evwilkin) [reply packages/patternfly-4/react-charts/src/components/ChartBullet/utils/chart-bullet-data.ts]: Yes the linter flags & changes these based on the [arrow-body-style rule](https://eslint.org/docs/rules/arrow-body-style)

File: packages/patternfly-4/react-charts/src/components/ChartBullet/utils/chart-bullet-size.ts (hunk 1 above)
Comment (evwilkin) [reply packages/patternfly-4/react-charts/src/components/ChartBullet/utils/chart-bullet-size.ts]: This was flagged by the [no-shadow](https://eslint.org/docs/rules/no-shadow) rule, as the function was named `scale` and created another variable inside it named `scale`.  It looks like the internal variable is pulled from props passed to this function, while the function itself is only called within the file and not exported, so changing the function name was easier than renaming the prop in this case.

File: packages/patternfly-4/react-charts/src/components/ChartBullet/utils/chart-bullet-theme.ts (hunk 3 above)
Comment (evwilkin) [reply packages/patternfly-4/react-charts/src/components/ChartBullet/utils/chart-bullet-theme.ts]: `index` comes from the `forEach` loop on line 29, and wasn't touched as part of this change.  This rule was new to me, you can find more on it here: [prefer-for-of](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-for-of.md)

File: packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts (hunk 7 above)
Comment (evwilkin) [reply packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts]: Thanks for the context - I'll add this back in!  It was originally removed per the [no-console](https://eslint.org/docs/rules/no-console) rule

File: packages/patternfly-4/react-charts/scripts/copyTS.js [hunk 11]
```diff
@@ -1,6 +1,8 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
 const path = require('path');
 const glob = require('glob');
 const fse = require('fs-extra');
+/* eslint-enable @typescript-eslint/no-var-requires */
```
Comment (dlabrecq) [inline packages/patternfly-4/react-charts/scripts/copyTS.js]: This is a build script. Not sure we need to apply linter here? Perhaps a different config for js?

File: packages/patternfly-4/react-charts/src/components/ChartDonutUtilization/ChartDonutUtilization.tsx (hunk 4 above)
Comment (evwilkin) [reply packages/patternfly-4/react-charts/src/components/ChartDonutUtilization/ChartDonutUtilization.tsx]: Reverted to original order

File: packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts (hunk 7 above)
Comment (evwilkin) [reply packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts]: Added console.error back in

File: packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts (hunk 8 above)
Comment (evwilkin) [reply packages/patternfly-4/react-charts/src/components/ChartUtils/chart-interactive-legend.ts]: Reverted function order

File: packages/patternfly-4/react-charts/src/components/ChartUtils/chart-legend.ts (hunk 9 above)
Comment (evwilkin) [reply packages/patternfly-4/react-charts/src/components/ChartUtils/chart-legend.ts]: Reverted function order

File: packages/patternfly-4/react-charts/src/components/ChartUtils/chart-theme.ts (hunk 10 above)
Comment (evwilkin) [reply packages/patternfly-4/react-charts/src/components/ChartUtils/chart-theme.ts]: Reverted function order

File: packages/patternfly-4/react-charts/scripts/copyTS.js (hunk 11 above)
Comment (evwilkin) [reply packages/patternfly-4/react-charts/scripts/copyTS.js]: Removed rule

**Pull request merged**